modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesRecoveryTask.java

```diff
@@ -169,23 +163,19 @@ private void reloadQueuesFromDB() throws AndesException {
 
 			queuesStored.removeAll(queueList);
 			for (AndesQueue queue : queuesStored) {
-				for (QueueListener listener : queueListeners) {
-					log.warn("Recovering node. Adding queue " + queue.toString());
-					/**
-					 * Ignoring MQTT queues when recovering as they are already stored in the database. 
-					 */
-					if (queue.getProtocolType() != ProtocolType.MQTT) {
-						listener.handleClusterQueuesChanged(queue, QueueListener.QueueEvent.ADDED);
-					}
+				log.warn("Recovering node. Adding queue " + queue.toString());
+				/**
+				 * Ignoring MQTT queues when recovering as they are already stored in the database.
+				 */
+				if (queue.getProtocolType() != ProtocolType.MQTT) {
```

Comment:
I could understand your reasons for doing this. But from a design perspective. having protocol handling logic in andes core is something we need to think about ? what are your thoughts ? @pamod @Asitha @a5anka @sirojan24 @pumudu88 @hastef88 @hemikak @sajinidesilva @indikasampath2000